test-integration/src/test/java/org/wso2am/micro/gw/tests/context/MgwServerInstance.java

```diff
@@ -0,0 +1,148 @@
+import org.wso2am.micro.gw.tests.util.HttpClientRequest;
+import org.wso2am.micro.gw.tests.util.HttpResponse;
+import org.wso2am.micro.gw.tests.util.Utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import static org.wso2am.micro.gw.tests.common.BaseTestCase.getMockServiceURLHttp;
+
+/**
+ * Mgw server instance class.
+ */
+public class MgwServerInstance implements MgwServer {
+
+    private static final Logger log = LoggerFactory.getLogger(MgwServerInstance.class);
+    private DockerComposeContainer environment;
+
+
+    /**
+     * initialize a docker environment using docker compose.
+     *
+     * @throws IOException
+     * @throws MicroGWTestException
+     */
+    public MgwServerInstance() throws IOException, MicroGWTestException {
+        createTmpMgwSetup();
+        File targetClassesDir = new File(MgwServerInstance.class.getProtectionDomain().getCodeSource().
+                getLocation().getPath());
+        String mgwServerPath = targetClassesDir.getParentFile().toString() + File.separator + "server-tmp";
+
+        String dockerCompsePath = mgwServerPath+  File.separator + "docker-compose.yaml";
+        MockBackendServer.addMockBackendServiceToDockerCompose(dockerCompsePath);
+        environment = new DockerComposeContainer(new File(dockerCompsePath))
+                .withLocalCompose(true);
+
+    }
+
+    /**
+     * initialize a docker environment using docker compose.
+     *
+     * @param confPath external conf.toml path
+     *
+     * @throws IOException
+     * @throws MicroGWTestException
+     */
+    public MgwServerInstance(String confPath) throws IOException, MicroGWTestException {
+        createTmpMgwSetup();
+        File targetClassesDir = new File(MgwServerInstance.class.getProtectionDomain().getCodeSource().
+                getLocation().getPath());
+        String mgwServerPath = targetClassesDir.getParentFile().toString() + File.separator + "server-tmp";
+        Utils.copyFile(confPath, mgwServerPath  +  File.separator + "resources"  +  File.separator +
+                "conf" +  File.separator + "config.toml");
+
+        String dockerCompsePath = mgwServerPath+  File.separator + "docker-compose.yaml";
+        MockBackendServer.addMockBackendServiceToDockerCompose(dockerCompsePath);
+        environment = new DockerComposeContainer(new File(dockerCompsePath))
+                .withLocalCompose(true);
+
+    }
+
+    @Override
+    public void startMGW() throws IOException, InterruptedException {
+        try {
+            environment.start();
+        } catch (Exception e) {
+            log.error("Error occurs when docker-compose up");
+        }
+
+        waitTillBackendIsAvailable();
+
+    }
+
+    @Override
+    public void stopMGW() {
+        environment.stop();
+    }
+
+    /**
+     * create a temporal mgw setup.
```

Comment:
Lets use proper comments

Reply:
Fixed in ee03466e7304d4b68a1c39906fb6a12affb17948